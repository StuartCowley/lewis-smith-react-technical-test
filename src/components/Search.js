// eslint-disable-next-line no-unused-vars
import React, { useState } from "react";
import "../styles/search.css";
import "../requests/getImages";
import getImages from "../requests/getImages";

const Search = ({ setSearchResults }) => {

    const [ value, setValue ] = useState();

    const handleSubmit = (event) => {
        event.preventDefault();
        setSearchResults(getImages(value));
    }

  return (
    <>
    <form className="search-form" onSubmit={handleSubmit}>
        <input 
            className="search-input"
            onChange={(e) => setValue(e.target.value)}
            type="text" 
        />
        <button className="search-button" type="submit">Search</button>
    </form>
   </>
   );
};

export default Search