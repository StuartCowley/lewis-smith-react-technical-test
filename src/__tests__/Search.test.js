import { render, screen } from '@testing-library/react';
import Search from '../components/Search';
import renderer from "react-test-renderer";

describe("Search", () => {
    it("renders correctly", () => {
        const validProps = {
            setSearchResult: jest.fn()
        }
        const rendered = renderer.create(<Search setSearchResult={validProps.setSearchResult}/>);

        expect(rendered).toMatchSnapshot();
    });

    it("Renders a single button with correct text", () => {
        const validProps = {
            setSearchResult: jest.fn()
        }
        render(
            <Search 
                setSearchResult={validProps.setSearchResult}
            />
        );

        const button = screen.getAllByRole("button");


        expect(button).toHaveLength(1);
        expect(button[0]).toHaveTextContent("Search");

    });
});