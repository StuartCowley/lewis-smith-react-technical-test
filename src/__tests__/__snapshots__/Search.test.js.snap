// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
<form
  className="search-form"
  onSubmit={[Function]}
>
  <input
    className="search-input"
    onChange={[Function]}
    type="text"
  />
  <button
    className="search-button"
    type="submit"
  >
    Search
  </button>
</form>
`;
